### PR TITLE
Cache Docker Desktop by build number

### PR DESCRIPTION
**Pull request by benja-M-1**
This PR does three things:
- it caches Docker Desktop per release ([3050dcb](https://github.com/docker/extensions-submissions/pull/3/commits/3050dcb37a365ae6e92da0844f2570d2413bb091))
- it refactors the workflows to use templates for comments everywhere
- it adds some "tests" that use [act](https://github.com/nektos/act) and [bats](https://github.com/bats-core/bats-core)

Screenshot 2023-01-20 at 00 32 55

Note: you should read this PR commit per commit

### CODE REVIEW

File: .github/workflows/validation.yaml [hunk 1]
```diff
@@ -8,55 +8,115 @@ env:
...
-          direction: last
+          body: ${{ steps.render.outputs.result }}
+
+      - name: Mark job as failed
+        run: exit 1
 
-      - name: Get Date
-        id: get-date
+  noop-validate:
+    if: github.event.act == true
```
Comment (benja-M-1) [inline .github/workflows/validation.yaml:63]: see https://github.com/nektos/act#skipping-jobs

File: .github/workflows/validation.yaml [hunk 2]
```diff
@@ -8,55 +8,115 @@ env:
...
+    steps:
+      - id: set-output
         run: |
-          echo "date=$(/bin/date -u "+%Y%m%d")" >> $GITHUB_OUTPUT
-        shell: bash
+          if [[ ${TEST_VALIDATION_FAILED} ]]; then
+            echo "validation_output=There are errors" >> $GITHUB_OUTPUT
+            exit 1
+          fi
+          if [[ ${TEST_VALIDATION_ERRORED} ]]; then
```
Comment (benja-M-1) [inline .github/workflows/validation.yaml:75]: This env var is populated by the `--env` option of `act` command line

```bash
act issues --env TEST_VALIDATION_FAILED=true -e test/issues/tos_accepted.json  --container-architecture linux/amd64
```

File: test/README.md [hunk 3]
```diff
@@ -0,0 +1,23 @@
+# How to test the workflows locally?
```
Comment (gtardif) [inline test/README.md:1]: Thx!

**Review by gtardif — APPROVED** [review]
Great!

**Pull request merged**